meeting-notes/2018/2018-06-25--js-core-dev-team-weekly.md

```diff
@@ -0,0 +1,157 @@
+# ⚡️ⒿⓈ Core Dev Team Weekly Sync 🙌🏽 June 25, 2018
+
+- **Lead:** Alan Shaw (@alanshaw)
+- **Notetaker:**
+- **Attendees:**
+  - @jacobheun
+  - @hugomrdias
+  - @fsdiogo
+  - @Mr0grog
+  - @achingbrain
+  - @vasco-santos
+  - @gar
+  - @vmx
+- **Recording:** _add link to recording once it's online_
+
+## Agenda
+
+- Ask everyone to put their name into the list of attendees
+- Round of updates
+  - What have you accomplished since the last Weekly?
+  - Were there any blockers? If so, which ones? Is it still blocked? Why?
+  - What is the next important thing you should focus on?
+- Ask for general questions. Could be things like:
+  - I'm stuck with something, I don't know who to ask. Who knows who to ask?
+  - Who can help me with xyz?
+- Plan this week
+  - Select issues to work on
+- Review remaining issues if there is time left
+- End of quarter actions
+  - Retrospective doc
+  - Scores for Q2 OKRs
+  - What are your objectives for next quarter?
+
+
+## Notes
+
+### Week Update
+
+@alanshaw
+  - Done:
+    - Investigated and raised an issue on go-ipfs for peer discovery regression on Windows [#5146](https://github.com/ipfs/go-ipfs/issues/5146)
+    - Added a WIP PR to expose libp2p connection manager config [#1410](https://github.com/ipfs/js-ipfs/pull/1410)
+    - Released new `js-ipfs-api` with fix for block get with empty blocks [#789](https://github.com/ipfs/js-ipfs-api/pull/789)
+    - Opened PR to address same block get issues in `js-ipfs` [#1408](https://github.com/ipfs/js-ipfs/pull/1408)
+    - Reviewed tweaked and merged PR for enabling pin tests [#1405](https://github.com/ipfs/js-ipfs/pull/1405)
+    - ...and more PR reviews and merges
+  - Blocked:
+    - `null`
+  - Next:
+    - Finish off remaining tasks for modular interface-ipfs-core tests [#290](https://github.com/ipfs/interface-ipfs-core/pull/290)
+    - Finish off remaining tasks for 0.30 release
+      - Connection manager merged ([#184](https://github.com/libp2p/js-libp2p/pull/184)) just needs a libp2p version bump
+      - By special request review and merge libp2p config PR [#166](https://github.com/libp2p/js-libp2p/pull/166)
+    - Review and merge MFS [#1360](https://github.com/ipfs/js-ipfs/pull/1360)
+
+@vmx
+ - Done:
+   - Get a full file via GraphSync hacks
+   - Review of IPLD Format changes implementation from @richardschneider
+ - Blocked: No
+ - Next:
+   - More reviews of IPLD Format changes implementation from @richardschneider
+   - Writing down what the essential core of GraphSync is from my perspective
+   - Playing around with js-libp2p with GraphSync in mind
+
+@hugomrdias
```

Comment:
@hugomrdias any advancements on the goals to load large files with js-ipfs?

Reply:
Fetching a large file will indeed be slow until GraphSync is implemented and SECIO perf is improved.

That said, it is important to have these tests reproducible. Is there a way to quickly do a test in Node.js to add 1TB to js-ipfs (it can be `head -c 1TB < /dev/urandom  | jsipfs add`) and one to do the same with 100GB in the browser (using similar strategy but with WebCrypto.randomBytes)?

Reply:
I can only speak for Node.js, not the Browser, but there GraphSync won't help much.

First of all, GraphSync might only make things faster when data isn't available on the peer it was requested from. From the demo it looks like testing was performed with requesting from a peer that has the data available. With Node.js the perf from getting it from such a peer seems reasonable (on my machine 800ms for a 100MiB file).

And even if the data is not available to the peer it was requested from, GraphSync won't make that much difference. On local testing it's like 2-3x difference to sending one big message via Bitswap (3-4s vs. 9s, although GraphSync does even less things than Bitswap). The difference within Bitswap between using one big message, or several smaller ones has waaay higher impact (8-9x, 9s vs. 1.2min).

Reply:
How deep is the graph you are testing with, @vmx?

Reply:
@vmx a file of 1TB is going to have a ton of depth, each level will be a Bitswap RTT. You need to do benchmarks for large files. It was expected that GraphSync won't do much for small files.

Reply:
@diasdavid: My point was that on the file sizes @hugomrdias is testing with, the magic bullet is not GraphSync, but fixing other bottlenecks.

Reply:
@vmx i have been testing with some huge files adding them on the daemon and fetching them in the browser, and i did see the bitswap round trips slowing it down but still nothing compares to crypto overhead.